microservices/realtime/src/Services/Sentinel.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+namespace Services;
+
+use Model\Message;
+use Model\RedisConf;
+use Swoole\Coroutine\Channel;
+
+class Sentinel
+{
+    const EVENT_SHUTDOWN = 'shutdown';
+
+    /** @var RedisConf[]  */
+    private $sentinels;
+
+    /** @var Channel  */
+    private $channel;
+
+    /** @var RedisConf */
+    private $master;
+
+    /** @var  RedisConf */
+    private $sentinel;
+
+    /**
+     * Sentinel constructor.
+     * @param RedisConf[] $sentinelConfig
+     */
+    public function __construct(
+        array $sentinelConfig
+    ) {
+        if (empty($sentinelConfig)) {
+            throw new \RuntimeException(
+                'Empty sentinel config found'
+            );
+        }
+
+        $this->sentinels = $sentinelConfig;
+        $this->channel = new Channel(1);
+    }
+
+    public function resolveMaster(): RedisConf
+    {
+        for ($i = 0; $i < count($this->sentinels); $i++) {
+            try {
+                $config = $this->sentinels[$i];
+                $this->master = $this->getRedisMasterOrThrowException(
+                    $config
+                );
+
+                $this->sentinel = $config;
+
+                break;
+            } catch (\Exception $e) {
+                echo "ERROR: " . $e->getMessage() . "\n";
+                continue;
```

Comment:
continue as last for statement is redundant but you can leave it as is if you want to clarify the differences between try and catch blocks